hr_timesheet_sheet/models/account_analytic_line.py

```diff
@@ -63,21 +63,28 @@ def _search_sheet(self, operator, value):
 
     @api.multi
     def write(self, values):
-        self._check_state()
+        self._check_state_on_write(values)
         return super(AccountAnalyticLine, self).write(values)
 
     @api.multi
     def unlink(self):
         self._check_state()
         return super(AccountAnalyticLine, self).unlink()
 
+    @api.multi
+    def _check_state_on_write(self, values):
+        """ Hook for extensions """
+        self._check_state()
+
+    @api.multi
     def _check_state(self):
+        if self._context.get('skip_check_state'):
```

Comment:
Use `self.env.context` instead

Reply:
Why is it preferable?

Reply:
That variable is internal and might change, and Odoo itself deprecated it: 

https://github.com/odoo/odoo/blob/5a1f874602ee649dfedf01c1be8f33d133677d00/odoo/models.py#L4354

Reply:
Thank you!